proposals/semi-auto-properties.md

```diff
@@ -10,6 +10,228 @@ Two common scenarios are that you want to apply a constraint on the setter, ensu
+
+Here's an example of where this is useful. The `field` keyword will find a lot of its use with view models because of the neat solution it brings for the `INotifyPropertyChanged` pattern. View model property setters are likely to be databound to UI and likely to cause change tracking or trigger other behaviors. The following code needs to initialize the default value of `IsActive` without setting `HasPendingChanges` to `true`:
+
+```cs
+using System.Runtime.CompilerServices;
+
+class SomeViewModel
+{
+    public bool HasPendingChanges { get; private set; }
+
+    public bool IsActive { get; set => Set(ref field, value); } = true;
+
+    private bool Set<T>(ref T location, T value)
+    {
+        if (RuntimeHelpers.Equals(location, value)) return false;
+        location = value;
+        HasPendingChanges = true;
+        return true;
+    }
+}
+```
+
+This difference in behavior between a property initializer and assigning from the constructor can also be seen with virtual auto-properties in previous versions of the language:
+
+```cs
+using System;
+
+// Nothing is printed; the property initializer is not
+// equivalent to `this.IsActive = true`.
+_ = new Derived();
+
+class Base
+{
+    public virtual bool IsActive { get; set; } = true;
+}
+
+class Derived : Base
+{
+    public override bool IsActive
+    {
+        get => base.IsActive;
+        set
+        {
+            base.IsActive = value;
+            Console.WriteLine("This will not be reached");
+        }
+    }
+}
+```
+
+### Constructor assignment
+
+As with existing auto-properties, assignment in the constructor calls the setter if it exists, and if there is no setter it falls back to directly assigning to the backing field.
+
+```cs
+class C
+{
+    public C()
+    {
+        P1 = 1; // Assigns P1's backing field directly
+        P2 = 2; // Assigns P2's backing field directly
+        P3 = 3; // Calls P3's setter
+    }
+
+    public int P1 => field;
+    public int P2 { get => field; }
+    public int P3 { get => field; set => field = value; }
+}
+```
+
+### Definite assignment in structs
+
+A semi-auto property will be treated as a regular auto property for the purposes of calculating default backing field initialization if its setter is automatically implemented, or if it does not have a setter ([LDM decision](https://github.com/dotnet/csharplang/blob/main/meetings/2022/LDM-2022-03-02.md#property-assignment-in-structs)).
+
+Default initialize a struct when calling a manually implemented semi-auto property setter, and issue a warning when doing so, like a regular property setter ([LDM decision](https://github.com/dotnet/csharplang/blob/main/meetings/2022/LDM-2022-05-02.md#definite-assignment-of-manually-implemented-setters)).
+
+### Nullability
```

Comment:
@RikkiGibson Would you be able to take a look at this section and tell me if there's more to do here? Thank you!

Reply:
I will try and think this through in more details. But off the cuff. I am thinking that we are going to have "gaps" here in nullable analysis which we will have to be comfortable with. We will have to err on the side of assuming things are non-null and assuming we shouldn't warn in order to avoid giving nuisance warnings which don't have an obvious way to be silenced.